R/alexanderZTest.R

```diff
@@ -0,0 +1,166 @@
+.z.test <- function(x, y=NULL, alternative="two.sided", mu=0, sigma.x=1, sigma.y=1, conf.level=0.95, 
+                    paired=FALSE, ciValueMeanDiffES=0.95) {
+  if (!is.null(y)) 
+    stop("Paired and two-sample z.test not yet implemented")
+  
+  result <- list(statistic=NULL, p.value=NULL, conf.int=NULL, estimate=NULL, null.value=mu, 
+                 alternative=alternative, method=NULL, data.name=NULL, 
+                 d=NULL, confIntEffSize=NULL)
+  
+  nX <- length(x)
+  nY <- NULL
+  
+  if (nX <= 1) 
+    stop("not enough x observations")
+  
+  xBar <- mean(x)
+  
+  if (is.null(y)) {
+    dname <- deparse(substitute(x))
+    
+    method <- "One-sample z-Test"
+    estimate <- xBar - mu
+    names(estimate) <- "mean difference of x"
+    d <- (xBar-mu)/sigma.x
+    zStat <- sqrt(nX) * d
+    sdError <- sigma.x/sqrt(nX)
+  } else {
+    dname <- paste(deparse(substitute(x)), "and", paste(deparse(substitute(y))))
+    
+    if (paired==TRUE) {
+      stop("Not yet implemented")
+      method <- "Paired z-test"
+    } 
+    
+    if (paired==FALSE) {
+      nY <- length(y)
+      
+      if (nY <= 1) 
+        stop("not enough y observations")
+      
+      yBar <- mean(y)
+      method <- "Two-sample z-Test"
+      estimate <- c(xBar, yBar)-mu
+      names(estimate) <- c("mean difference of x", "mean difference of y")
+      
+      sdError <- sqrt(((sigma.x^2)/nX) + ((sigma.y^2)/nY))
+      zStat <- (xBar - yBar - mu)/stderr
+      nEff <- (1/nX + 1/nY)^(-1)
+      d <- zStat/sqrt(nEff)
+    }
+  }
+  
+  # TODO(Alexander): Doesn't work for two-sample problems. Furthermore, only scaled wrt sigma.x
+  # 
+  # args(.zMeanDifferenceConfidenceIntervalStat)
+  
+  
+  tempResult <- .zMeanDifferenceConfidenceIntervalStat("zStat"=zStat, "ciValueMeanDiff"=conf.level,
+                                                       "sdError"=sdError, "nX"=nX, "nY"=nY, "alternative"=alternative,
+                                                       "ciValueMeanDiffES"=ciValueMeanDiffES)
```

Comment:
why the "zStat" here? you removed it in the other locations (which I like, to be honest).